src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -15,6 +15,11 @@ public partial struct AbsolutePath : IEquatable<AbsolutePath>, IPath
 
     private static readonly string DirectorySeparatorCharStr = Path.DirectorySeparatorChar.ToString();
 
+    /// <summary>
+    /// File system implementation.
+    /// </summary>
+    public IFileSystem FileSystem { get; set; }
```

Comment:
This increases item size by `sizeof(nint)`; which puts the struct at `3x(sizeof(nint))` size total; basically the boundary between when it's worth to still use a struct vs. a class.

Given in the actual app we're most likely going to use `FileSystem` directly [as `InMemoryFileSystem` is mostly a testing/debugging tool]; we'll end up copying the same instance across each path item; which is a bit unoptimal. 

For now I think this is okay. Down the road though, I think ideally we'd just have extension methods on `IFileSystem` or on `AbsolutePath` which perform an operation using the path; such that memory wise `IFileSystem` can be decoupled from `AbsolutePath`.

Reply:
This is mostly just a stopgap measure to prevent this PR from breaking everything. Once newer code uses `IFileSystem` and we update the older code, we can remove this property from `AbsolutePath`, as well as all the utility methods in `AbsolutePath` which use `IFileSystem` (eg: `AbsolutePath.ReadAllTextAsync` just redirects to `FileSystem.ReadAllTextAsync`).

Reply:
Yeah; just remember, the later it's done the harder it'll be to migrate 😉